Framework/DataObjects/src/EventList.cpp

```diff
@@ -3689,11 +3685,9 @@ void EventList::filterByTimeAtSampleHelper(std::vector<T> &events, DateAndTime s
   while ((itev != itev_end) && (calculateCorrectedFullTime(*itev, tofFactor, tofOffset) < start.totalNanoseconds()))
     itev++;
 
-  while ((itev != itev_end) && (calculateCorrectedFullTime(*itev, tofFactor, tofOffset) < stop.totalNanoseconds())) {
-    // Add the copy to the output
-    output.emplace_back(*itev);
-    ++itev;
-  }
+  std::copy_if(itev, itev_end, std::back_inserter(output), [stop, tofFactor, tofOffset](T &t) {
+    return calculateCorrectedFullTime(t, tofFactor, tofOffset) < stop.totalNanoseconds();
+  });
```

Comment:
same as the previous one, unless there are performance concerns, in which case it needs to be stated as comments here.